### DISCUSSION TITLE
Errors and poor performance using fastApi/uvicorn compared to falcon/gunicorn

### DISCUSSION THREAD

**Discussion by Digitalean** [fastapi/fastapi#8729] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
def get_app() -> FastAPI:
    mycompnent_app = FastAPI(
        title="mycompnent",
        description="my app",
        openapi_tags=MYCOMPONENT_METADATA
    )
    mycompnent_app.include_router(api_routeur, prefix=MYCOMPONENT_PREFIX)
    mycompnent_app.add_event_handler("startup", start_app_handler(mycompnent_app))
    mycompnent_app.add_event_handler("shutdown", stop_app_handler(mycompnent_app))

    return mycompnent_app

app = get_app()

---router--
@router_m.post("/predict", response_model=ObjectEnrichiInference, name="prediction" )
async def prediction(object: ObjectDTO, request: Request) -> ObjectEnrichiInference:
    .....  
    return msg
```

### Description

Hello,
We have a machine learning use case where using uvicorn with fast api has bad performances.
You can find the results of the test via this link: (https://digitalean.fr/performance_python_machine_learning_use_case.pdf)

Can you tell us why we have this performance please? And can you help us to have better performance with uvicorn and fast api please?

Thanks in advance.

### Operating System

Other

### Operating System Details

Kubernetes python pods 

### FastAPI Version

fastapi==0.65.1 uvicorn[standard]==0.13.4

### Python Version

python:3.7

### Additional Context

Hello,
We have a machine learning use case where using uvicorn with fast api has bad performances.
You can find the results of the test via this link: (https://digitalean.fr/performance_python_machine_learning_use_case.pdf)

Can you tell us why we have this performance please? And can you help us to have better performance with uvicorn and fast api please?

Thanks in advance.

**Comment by jgould22** [comment]
Highly likely you are blocking the event loop with whatever you are doing in `prediction`

gunicorn/gthread is mitigating this for you by putting things in a thread pool.

If you change your `async def prediction` to `def prediction` FastAPI will spawn it on a thread pool and things might speed up a bit  but you should not be doing intense computational work in async python because it is cooperative not preemptive multitasking.

**Comment by adriangb** [comment]
Or maybe your workload is just not async (most ML workloads are CPU bound) so you're comparing the overhead of an async framework with a better performing (via Cython and less features) sync framework
